WordPress/gutenberg
# [ListView] Allow deleting blocks using keyboard

## What?

Fix https://github.com/WordPress/gutenberg/issues/49087. Allow deleting blocks using the `Delete` and `Backspace` key, or the keyboard shortcut `Shift+Alt+z` (`Ctrl+Alt+z` on macOS).

## Why?


To retain feature parity. See https://github.com/WordPress/gutenberg/issues/49087.

## How?

Added a `removeRow` function and a `KeyboardShortcuts` component.

## Testing Instructions




1. Open a Post and insert some blocks.
2. Open the list view.
3. Try using the keyboard to navigate and focus/select some blocks in the list view
4. Press `Delete`, `Backspace`, or `Shift+Alt+z` (`Ctrl+Alt+z` on macOS) to delete the focused blocks.
5. Expect the focused blocks to be deleted.
6. Try other things like deleting locked blocks or deleting unselected but focused blocks and see if it works as expected.

### Testing Instructions for Keyboard

1. Open a Post and insert some blocks.
2. Open the list view (`Shift+Alt+o` or `Ctrl+Alt+o` on macOS).
3. Try using the keyboard to navigate and focus/select some blocks in the list view
4. Press `Delete`, `Backspace`, or `Shift+Alt+z` (`Ctrl+Alt+z` on macOS) to delete the focused blocks.
5. Expect the focused blocks to be deleted and expect the focus to be retained in the list view (rather than moving to the canvas).
6. Try other things like deleting locked blocks or deleting unselected but focused blocks and see if it works as expected.

## Screenshots or screencast 


https://user-images.githubusercontent.com/7753001/236965793-2510d588-3655-4622-90b1-652052ce608a.mp4
